scan-command/src/main/java/io/ballerina/scan/internal/ScanCmd.java

```diff
@@ -131,13 +133,23 @@ public void execute() {
             return;
         }
 
-        ProjectAnalyzer projectAnalyzer = new ProjectAnalyzer();
-        List<Rule> coreRules = CoreRule.rules();
+        if (ProjectUtils.isProjectEmpty(project.get())) {
+            outputStream.println("ballerina: package is empty. Please add at least one .bal file.");
```

Comment:
Thinking out loud, even for other commands, I'm not entirely sure if this is a valid ask, since adding an empty bal file will make this error go away. We can rethink this separately.